### PR TITLE
Http client docs

### PR DESCRIPTION
**Pull request by ZacharyKlein**

### CODE REVIEW

File: src/en/guide/REST/callingRestServices.adoc [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+There are a few ways by which you can obtain a reference to a https://docs.micronaut.io/latest/api/io/micronaut/http/client/HttpClient.html[HttpClient]. The most simple way is using the https://docs.micronaut.io/latest/api/io/micronaut/http/client/HttpClient.html#create-java.net.URL-[create] method
+
+.Creating an HTTP client
+[source, groovy]
+----
+    List<Album> searchWithApi(String searchTerm) {
+        String baseUrl = "https://itunes.apple.com/"
+
+        HttpClient client = HttpClient.create(baseUrl.toURL()).toBlocking() //<1>
```
Comment (graemerocher) [inline src/en/guide/REST/callingRestServices.adoc]: The client should be defined as a spring bean and shutdown with the `close` method. Not closing the client will leak threads.

**Comment by ZacharyKlein** [comment]
@graemerocher could you clarify this? `The client should be defined as a spring bean`

As far as I know, the HTTP client has to be created via the `create` method, not a constructor call. I'm not sure how to define a Spring bean (e.g., in `resources.groovy`) where a method other than the constructor is being used. Or am I misunderstanding your statement?

**Comment by graemerocher** [comment]
@ZacharyKlein how to define a bean that is created via a factory method is described in the Grails docs http://docs.grails.org/latest/guide/spring.html#theBeanBuilderDSLExplained

**Comment by ZacharyKlein** [comment]
@graemerocher I experimented (with some help from James) and this is the closest I can get to defining a `HttpClient` bean: ```beans = {
    httpClient(HttpClient) { bean ->
        bean.factoryMethod = "create"
        bean.constructorArgs = ["http://localhost:8080/api/widget".toURL()]
        bean
    }
}
```

However, this fails to start as a required bean of type `io.micronaut.context.BeanContext` cannot be found. Is there a way to get a ref of the Micronaut parent context and make that available to the `httpClient` bean? Or am I going about this wrong?

**Pull request merged**